src/EditorFeatures/CSharp/RawStringLiteral/RawStringLiteralCommandHandler_Return.cs

```diff
@@ -50,12 +51,164 @@ public bool ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext co
+        // We could reach this point after pressing enter on a single-line raw string
+
+        var currentSnapshot = subjectBuffer.CurrentSnapshot;
+        var document = currentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+        if (document == null)
             return false;
 
+        var parsedDocument = ParsedDocument.CreateSynchronously(document, cancellationToken);
+
+        var token = parsedDocument.Root.FindToken(position);
+        switch (token.Kind())
+        {
+            case SyntaxKind.SingleLineRawStringLiteralToken:
+                break;
+
+            case SyntaxKind.InterpolatedStringTextToken
+            when token.Parent?.Parent is InterpolatedStringExpressionSyntax interpolated &&
+                interpolated.StringStartToken.Kind() is SyntaxKind.InterpolatedSingleLineRawStringStartToken:
+                break;
+
+            default:
+                return false;
+        }
+
+        var indentationOptions = subjectBuffer.GetIndentationOptions(_editorOptionsService, document.Project.GetFallbackAnalyzerOptions(), document.Project.Services, explicitFormat: false);
+        var indentation = token.GetPreferredIndentation(parsedDocument, indentationOptions, cancellationToken);
+
+        var newLine = indentationOptions.FormattingOptions.NewLine;
+
+        using var transaction = CaretPreservingEditTransaction.TryCreate(
+            CSharpEditorResources.Split_raw_string, textView, _undoHistoryRegistry, _editorOperationsFactoryService);
+
+        var edit = subjectBuffer.CreateEdit();
+
+        // Add a newline at the position of the end literal
+        var closingStart = GetStartPositionOfClosingDelimiter(token);
+        var newLineAndIndentation = newLine + indentation;
+        var insertedLines = 1;
+        edit.Insert(closingStart, newLineAndIndentation);
+        // Add a newline at the requested position
+        edit.Insert(position, newLineAndIndentation);
+        // Also add a newline at the start of the text, only if there is text before the requested position
+        var openingEnd = GetEndPositionOfOpeningDelimiter(token);
+        if (openingEnd != position)
+        {
+            insertedLines++;
+            edit.Insert(openingEnd, newLineAndIndentation);
+        }
+        var snapshot = edit.Apply();
+
+        // move caret:
+        var lineInNewSnapshot = snapshot.GetLineFromPosition(openingEnd);
+        var nextLine = snapshot.GetLineFromLineNumber(lineInNewSnapshot.LineNumber + insertedLines);
+        textView.Caret.MoveTo(new VirtualSnapshotPoint(nextLine, indentation.Length));
+
+        transaction?.Complete();
+        return true;
+    }
+
+    private static int GetEndPositionOfOpeningDelimiter(SyntaxToken currentStringLiteralToken)
+    {
+        switch (currentStringLiteralToken.Kind())
+        {
+            case SyntaxKind.SingleLineRawStringLiteralToken:
```

Comment:
When do we have the multi line case?

Reply:
Those are leftovers, will remove
This command doesn't handle multi-line raw strings and they shouldn't be expected in these methods either

Reply:
Multiline cases are now supported because of the underlying generalization

Reply:
i don't know what that means.  it looks like earlier up we bail out in the multi-line case.